engine/src/test/java/com/sandboni/core/engine/sta/SeloniJsonConnectorTest.java

```diff
@@ -0,0 +1,106 @@
+package com.sandboni.core.engine.sta;
+
+import com.sandboni.core.engine.contract.HttpConsts;
+import com.sandboni.core.engine.sta.connector.Connector;
+import com.sandboni.core.engine.sta.connector.HttpTemplateConnector;
+import com.sandboni.core.engine.sta.graph.Link;
+import com.sandboni.core.engine.sta.graph.LinkFactory;
+import com.sandboni.core.engine.sta.graph.LinkType;
+import com.sandboni.core.engine.sta.graph.vertex.CucumberVertex;
+import com.sandboni.core.engine.sta.graph.vertex.TestVertex;
+import com.sandboni.core.engine.sta.graph.vertex.Vertex;
+import com.sandboni.core.scm.scope.ChangeScopeImpl;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import static org.junit.Assert.*;
+
+public class SeloniJsonConnectorTest {
```

Comment:
Just to confirm, are you covering all the possible scenarios for this connector? seems complex with multiple branches. Let's create as many tests as possible to avoid any "manual" testing

Reply:
added